lib/cocoapods/generate/podfile_generator.rb

```diff
@@ -123,9 +123,16 @@ def inhibit_all_warnings?
       #         whether all pods should use modular headers
       #
       def use_modular_headers?
-        return false unless configuration.use_podfile?
-        target_definition_list.all? do |target_definition|
-          target_definition.use_modular_headers_hash['all']
+        if configuration.use_podfile? && configuration.use_modular_headers?
+          raise 'Conflicting `use_modular_headers` option. Cannot specify both `--use-modular-headers` and `--use-podfile`.'
```

Comment:
`raise  Informative, ...`

Reply:
That makes this a non-error though. Is that what you'd prefer?

Reply:
It will still exit with 1 but also theme the message red etc. Its standard practice for user errors in cocoapods

Reply:
adding `Informative` makes this non catchable by the typical `expect().to raise_error` and I don't see much else on the webs about `Informative` errors

Reply:
ping @segiddins @dnkoutso . What would you prefer me to do here? Informative isn't as far as I can tell a catchable Rspec error.